spec/PhpSpec/Runner/Maintainer/ErrorMaintainerSpec.php

```diff
@@ -46,6 +46,19 @@ function it_returns_false_when_error_suppressed_in_php_8()
         error_reporting($oldLevel);
     }
 
+    function it_returns_false_when_error_is_a_deprecation()
+    {
+        /**
+         * Based on manual testing and:
+         * @link https://www.php.net/manual/en/language.operators.errorcontrol.php#125938
+         */
+        $oldLevel = error_reporting(E_DEPRECATED);
+
+        $this->errorHandler(E_DEPRECATED, 'error message', 'file', 1)->shouldBe(false);
+
+        error_reporting($oldLevel);
```

Comment:
you you need to put this in a try-finally.

Reply:
what do you mean? we are on the spec file here and this is the same way as on the method just above.

https://github.com/phpspec/phpspec/blob/main/spec/PhpSpec/Runner/Maintainer/ErrorMaintainerSpec.php#L44